misc/python/materialize/buildkite.py

```diff
@@ -40,9 +40,74 @@ def get_pipeline_default_branch(fallback: str = "main"):
     return os.getenv("BUILDKITE_PIPELINE_DEFAULT_BRANCH", fallback)
 
 
-def get_merge_base(remote="origin") -> str:
+def get_remote(url: str) -> str | None:
+    result = spawn.capture(["git", "remote", "--verbose"])
+    for line in result.splitlines():
+        remote, desc = line.split("\t")
+        if desc.lower() == f"{url} (fetch)".lower():
+            return remote
+    return None
+
+
+def get_merge_base(url: str = "https://github.com/MaterializeInc/materialize") -> str:
+    # Alternative syntax
+    remote = get_remote(url) or get_remote(
+        url.replace("https://github.com/", "git@github.com:")
+    )
+    if not remote:
+        print(f"Remote for URL {url} not found, using origin")
+        remote = "origin"
     base_branch = get_pull_request_base_branch() or get_pipeline_default_branch()
     merge_base = git.get_common_ancestor_commit(
         remote, branch=base_branch, fetch_branch=True
     )
     return merge_base
+
+
+def inline_link(url: str, label: str | None = None) -> str:
+    """See https://buildkite.com/docs/pipelines/links-and-images-in-log-output"""
+    link = f"url='{url}'"
+
+    if label:
+        link = f"{link};content='{label}'"
+
+    # These escape codes are not supported by terminals
+    return f"\033]1339;{link}\a" if is_in_buildkite() else f"{label},{url}"
+
+
+def inline_image(url: str, alt: str) -> str:
+    """See https://buildkite.com/docs/pipelines/links-and-images-in-log-output#images-syntax-for-inlining-images"""
+    content = f"url='\"{url}\"';alt='\"{alt}\"'"
+    # These escape codes are not supported by terminals
+    return f"\033]1338;{content}\a" if is_in_buildkite() else f"{alt},{url}"
+
+
+def find_modified_lines() -> set[tuple[str, int]]:
+    """
+    Find each line that has been added or modified in the current pull request.
+    """
+    merge_base = get_merge_base()
+    print(f"Merge base: {merge_base}")
+    result = spawn.capture(["git", "diff", "-U0", merge_base])
```

Comment:
@def-: Are you sure we want to use `-U0` here? I am not.
I would rather prefer that we add a couple of lines as context. One should not need to change the TODO comment for this to trigger; instead, it should be enough to conduct a close-by change.

Reply:
My intention with this was to prevent people from breaking nightly. If they only touched surrounding lines their change won't break nightly.